units/systemd-nspawn@.service.in

```diff
@@ -13,7 +13,7 @@ Before=machines.target
 After=network.target
 
 [Service]
-ExecStart=@bindir@/systemd-nspawn --quiet --keep-unit --boot --link-journal=try-guest --network-veth --settings=override --machine=%i
+ExecStart=@bindir@/systemd-nspawn --quiet --keep-unit --boot --link-journal=try-guest --network-veth -U --settings=override --machine=%i
```

Comment:
This is actually on purpose -U is actually slightly different from --private-users=pick. -U will downgrade gracefully when the kernel doesn't support user namespaces, but --private-users=pick will fail. Thus, we can safely add -U to the .service file and maintain compat with userns supporting and userns-less systems. If we'd use --private-users=pick instead, we'd start failing on systems where userns is not supported. Also see the man page for this.

BTW, I named this --private-users=pick instead of --private-users=auto on purpose. I wanted to leave the --private-users=auto thing free, so that we can use it one day in case the kernel lerns uid mapping mounts in some fashion.